File: tic-tac-toe-core/src/main/java/by/xxx/pupil/ai/minimax/findmoves/DefenseMovesFinder.java
package by.xxx.pupil.ai.minimax.findmoves;

import by.xxx.pupil.model.Board;
import by.xxx.pupil.model.Move;
import by.xxx.pupil.model.Player;
import org.apache.commons.lang3.Validate;

import java.util.List;

public class DefenseMovesFinder implements MovesFinder {

    private final InRadiusMovesFinder inRadiusMovesFinder;

    public DefenseMovesFinder(InRadiusMovesFinder inRadiusMovesFinder) {
        Validate.notNull(inRadiusMovesFinder, "null inRadiusMovesFinder");

        this.inRadiusMovesFinder = inRadiusMovesFinder;
    }

    @Override
    public List<Move> getMoves(Board board, Player player) {
        List<Move> moves = inRadiusMovesFinder.getMoves(board, player);


        return null;
    }

}

File: tic-tac-toe-core/src/main/kotlin/by/xxx/pupil/ai/minimax/findmoves/DefenseMovesFinder.kt
package by.xxx.pupil.ai.minimax.findmoves

import by.xxx.pupil.model.Board
import by.xxx.pupil.model.Move
import by.xxx.pupil.model.Player

class DefenseMovesFinder(private val inRadiusMovesFinder: InRadiusMovesFinder) : MovesFinder {

    override fun getMoves(board: Board, player: Player): List<Move> {
        val moves = inRadiusMovesFinder.getMoves(board, player)
        TODO("Complete moves finder")
    }

}